### PR TITLE
Feat: Add gpt 4o blog + LangX case study + typo fix

### LINKED ISSUE #975: 📚 Documentation: Small typo in the Appwrite vs Firebase blog
### 💭 Description

In the resources section of the [Appwrite vs Firebase blog](https://appwrite.io/blog/post/open-source-firebase-alternative), there is an `e` missing in `Appwrite YouTube`

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by adityaoberai**
## What does this PR do?

- Add GPT-4o x Appwrite Functions blog
- Add LangX case study
- Fix typo in Firebase blog

## Test Plan

`pnpm i && pnm run dev`

## Related PRs and Issues

Fixes #975 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by gewenyu99 — APPROVED** [review]
Random comment, looks good but can you do a drop in replace for `’` to be `'`?

**Comment by gewenyu99** [comment]
Screenshot 2024-05-21 at 9 49 04 AM
Also these images should be framed or zoomed.

The rule we have is:
- If we can zoom intentionally to focus on important parts, zoom.

If full page screenshot, it needs to be framed using Brandbird. Ask @chenparnasa for details :D

**Comment by adityaoberai** [comment]
> Random comment, looks good but can you do a drop in replace for `’` to be `'`?

Done this @gewenyu99 👍

**Comment by adityaoberai** [comment]
> Screenshot 2024-05-21 at 9 49 04 AM Also these images should be framed or zoomed.
> The rule we have is:
> 
> * If we can zoom intentionally to focus on important parts, zoom.
> 
> If full page screenshot, it needs to be framed using Brandbird. Ask @chenparnasa for details :D

Updated the images based on the guidelines sent by Chen

File: src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+Therefore, in this blog, we will learn how to use Appwrite Functions and the OpenAI GPT-4o API to build a personal chatbot.
+
+## Setting up the OpenAI Platform
+
+To use the OpenAI GPT-4o API, we must first create an API key on their platform. To do that, we must first create an account on the [OpenAI platform](https://platform.openai.com/). Once the account is set up and a project is created, we can visit their [API keys](https://platform.openai.com/account/api-keys) page and create an API key. Ensure you copy and save this key in a safe place, as the OpenAI platform will not let you view the key after it is created.
+
+![Create an OpenAI API Key](/images/blog/personal-chatbot-gpt-4o/openai.png)
+
+> Note: To use the GPT-4o API, your account must be upgraded to the Usage tier 1 (you must purchase at least $5 worth of credits). To learn more, visit their [Usage tiers documentation](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one).
```
Comment (stnguyen90) [inline src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc:22]: "the Usage tier 1" sounds a bit odd to me. Maybe:

```suggestion
> Note: To use the GPT-4o API, your account must be upgraded to Usage tier 1 (you must purchase at least $5 worth of credits). To learn more, visit their [Usage tiers documentation](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one).
```

or

```suggestion
> Note: To use the GPT-4o API, your account must be upgraded to the Tier 1 Usage tier (you must purchase at least $5 worth of credits). To learn more, visit their [Usage tiers documentation](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one).
```

File: static/images/blog/personal-chatbot-gpt-4o/cover.png
Comment (stnguyen90) [inline static/images/blog/personal-chatbot-gpt-4o/cover.png:1]: This spacing or so with this cover looks off. Did design approve this?

File: src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+### Testing the function
+
+Once you've completed all the aforementioned steps, you can push the code to our GitHub repository, at which point Appwrite Cloud will automatically deploy the changes to your function.
+
+![Appwrite Function deployments](/images/blog/personal-chatbot-gpt-4o/deployment.png)
+
+You can then go ahead and test your function by opening the function domain in your browser.
+
+Here is what an example of this looks like: [apwr.dev/gpt-4o-demo](https://apwr.dev/gpt-4o-demo)
```
Comment (stnguyen90) [inline src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc:264]: How are we going to afford to keep this online 🧐 ?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc (hunk 1 above)
Comment (adityaoberai) [reply src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc:22]: @stnguyen90 `Usage tier 1` is language taken from their documentation

File: static/images/blog/personal-chatbot-gpt-4o/cover.png
Comment (adityaoberai) [reply static/images/blog/personal-chatbot-gpt-4o/cover.png:1]: The image was provided as an asset for our AI posts from the Design team

I confirmed with Laura before adding this here

File: src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc (hunk 2 above)
Comment (adityaoberai) [reply src/routes/blog/post/personal-chatbot-gpt-4o/+page.markdoc:264]: Based on usage seen on past similar projects, I have more than enough credits available to keep this running for a while

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**